openssl/src/crypto/rsa.rs

```diff
@@ -61,6 +74,19 @@ impl RSA {
         }
     }
 
+    /// set the key modulus
+    pub fn set_n(&mut self, n: BigNum) {
+        unsafe {
+            (*self.0).n = n.into_raw();
```

Comment:
Should this be taking any locks on the OpenSSL side since `RSA`s are ref-counted?